include/boost/numeric/ublas/tensor/static_strides.hpp

```diff
@@ -267,18 +266,22 @@ struct basic_static_strides<basic_static_extents<T,Extents...>, Layout>
     return m_data.empty();
   }
 
-  template<class OtherE>
-  constexpr bool operator==(basic_static_strides<OtherE,layout_type> const& rhs) const noexcept{
-    if constexpr( _size != basic_static_strides<OtherE,layout_type>::_size ){
-      return false;
+  template <class Strides, std::enable_if_t<is_strides_v<Strides>, int> = 0 >
+  [[nodiscard]] inline
+  constexpr bool operator==(Strides const& rhs) const noexcept{
+    static_assert(is_strides_v<Strides>,
+        "boost::numeric::ublas::operator==() : invalid type, type should be an extents");
+    if( this->size() != rhs.size() ){
+        return false;
     }else{
-      return std::equal(begin(), end(), rhs.begin());
+        return std::equal(this->begin(), this->end(), rhs.begin());
```

Comment:
I recommend to simplify the statement here:
```cpp
return this->size() == rhs.size() && return std::equal(this->begin(), this->end(), rhs.begin());
```

Reply:
same as [here](https://github.com/boostorg/ublas/pull/81#discussion_r417212093)